internal/sapsystem/sapsystem.go

```diff
@@ -298,24 +275,19 @@ func getDBAddress(system *SAPSystem) (string, error) {
 	return "", fmt.Errorf("could not get any IPv4 address")
 }
 
-func setSystemID(fs afero.Fs, system *SAPSystem) (*SAPSystem, error) {
-	// Set system ID
-	var err error
-	var id string
-
-	switch system.Type {
+func detectSystemID(fs afero.Fs, executor utils.CommandExecutor, sType SystemType, sid string) (string, error) {
+	switch sType {
 	case Database:
-		id, err = getUniqueIDHana(fs, system.SID)
+		return getUniqueIDHana(fs, sid)
 	case Application:
-		id, err = getUniqueIDApplication(system.SID)
+		return getUniqueIDApplication(executor, sid)
 	case DiagnosticsAgent:
-		id, err = getUniqueIDDiagnostics(fs)
+		return getUniqueIDDiagnostics(fs)
 	case Unknown:
-		id = "-"
+		return "-", nil
```

Comment:
We could use a `default` here and avoid having the `return "-", nil` repeated again on the next line